### PR TITLE
adding abillity to define services imperative

### PR DESCRIPTION
**Pull request by delijati**
See issue #334 for more info. Coverage for sphinxext increased to 60% :)

### CODE REVIEW

File: cornice/resource.py [hunk 1]
```diff
@@ -24,83 +24,127 @@ def resource(depth=1, **kw):
...
+    All the methods of this class named by the name of HTTP resources
+    will be used as such. You can also prefix them by "collection_" and they
+    will be treated as HTTP methods for the given collection path
+    (collection_path), if any.
+
+    Here is an example::
+        class User(object):
+            pass
+
+        add_resource(User, collection_path='/users', path='/users/{id}')
```
Comment (ametaireau) [inline cornice/resource.py]: If possible, add the `:params:` syntax which explains what is contained in the arguments.

File: docs/source/resources.rst [hunk 2]
```diff
@@ -30,6 +30,24 @@ Here is how you can register a resource::
             _USERS[len(_USERS) + 1] = self.request.json_body
             return True
 
+Here a example of imperative way defining a resource::
```
Comment (ametaireau) [inline docs/source/resources.rst]: consider: "Here is an example of how to define cornice resources in an imperative way"

File: docs/source/resources.rst [hunk 3]
```diff
@@ -30,6 +30,24 @@ Here is how you can register a resource::
...
+
+    from cornice.resource import add_resource, add_view
```
Comment (ametaireau) [inline docs/source/resources.rst]: I would rather import `resource` and use `resource.add_view` and `resource.add_resource`.

**Comment by ametaireau** [comment]
Hi!

Thanks for taking the time to submit this pull request. That's appreciated. I like how the decorators are now completely optional, and in some way, it closes the gap I started to fill when I did [the cornice refactor](http://blog.notmyidea.org/refactoring-cornice.html).

I'm not a big fan of the `add_resource` and `add_view` names, but they seem to match with the ones provided by pyramid itself, so I think it's *fine* to let it like that.

I've added a few comments and I think we're good for merging once they're addressed.

**Comment by delijati** [comment]
Hi,

i've added the directive `add_cornice_resource` to get rid of the `scan`. There is still that issue with ordering of the resource views. I think a cleaner way would be to add a `Register` Object like the Service object witch holds the collected informations. But this should be done in another refactoring ;)

**Comment by delijati** [comment]
Do you had some time to look into that?

**Comment by delijati** [comment]
@leplatrem could you also take a look into that pull request

**Comment by ametaireau** [comment]
Hey Josip, this looks really good. Eveything is in there: impelmentation, docs, new tests, plus the old tests are passing.

Let's merge :-)

**Comment by delijati** [comment]
Wonderful do you know if someone is already working on the push request https://github.com/mozilla-services/cornice/pull/285 ? If not i will look into that one next week.

**Comment by ametaireau** [comment]
Neat. Nobody is currently working on this I believe. Feel free to take it!

**Pull request merged**